### PR TITLE
[FLINK-35391][cdc-connector][paimon] Bump dependency of Paimon Pipeline connector to 0.8.0

### PR DESCRIPTION
**Pull request by beryllw**
fix [FLINK-35391](https://issues.apache.org/jira/browse/FLINK-35391)

### CODE REVIEW

**Comment by beryllw** [comment]
@lvyanquan cc, thanks.
Paimon 0.8.0 use dynamic bucket as default, and FLINK-35394 prepare to support dynamic bucket. For bump paimon to 0.8.0, we can set unit test paimon table bucket default 1 first.

**Comment by beryllw** [comment]
PS: An error will be reported during local testing with a lower version of jdk (like jdk 1.8.0). Does it need to be fixed?
```
[ERROR] Error: VM option 'UseG1GC' is experimental and must be enabled via -XX:+UnlockExperimentalVMOptions.
```

File: pom.xml [hunk 1]
```diff
@@ -421,6 +421,7 @@ limitations under the License.
                         <exclude>**/.*/**</exclude>
                         <!-- Generated content -->
                         <exclude>**/target/**</exclude>
+                        <exclude>**/*.log</exclude>
```
Comment (lvyanquan) [inline pom.xml:424]: What is the reason for the generation of these logs?

**Comment by beryllw** [comment]
> Does this error only affect local testing and does not affect running with jar? If it does not affect the operation after packaging, it can be ignored.

OK, just affect local testing.

File: pom.xml (hunk 1 above)
Comment (beryllw) [reply pom.xml:424]: `Derby.log` will be generated when using idea for paimon sink local testing. 
`flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/derby.log`

**Review by lvyanquan — APPROVED** [review]
LGTM.

**Review by yuxiqian — APPROVED** [review]
LGTM

**Comment by lvyanquan** [comment]
Considering this optimization, upgrading the payment version is necessary.
https://github.com/apache/paimon/pull/2943

**Comment by yuxiqian** [comment]
Paimon team has released 8.0.1 patch version last month [here](https://paimon.apache.org/downloads/). Maybe we can upgrade to this version without encountering any compatibility issues.

**Comment by yuxiqian** [comment]
Could @leonardBang help driving this? Just noticed that Paimon 0.8.2 got released...

**Comment by leonardBang** [comment]
@beryllw Would you like to bump paimon version to 8.0.2?

**Comment by beryllw** [comment]
> @beryllw Would you like to bump paimon version to 8.0.2?

done(paimon 0.8.2)

**Comment by beryllw** [comment]
Looks like an error related to MySQL CDC.
```2024-07-16T10:37:08.5109144Z 52153 [debezium-reader-1] ERROR io.debezium.relational.TableSchemaBuilder - Failed to properly convert data value for 'date_convert_test_370b26.date_convert_test.test_time' of type TIME for row [1, 2023-04-01T14:23:00.000+0000, 2023-04-01T14:24:00.000+0000, 2023-04-01, PT14H25M]:
2024-07-16T10:37:08.5111918Z java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: YearOfEra```

**Comment by lvyanquan** [comment]
@czy006 PTAL as this failed case maybe related to https://github.com/apache/flink-cdc/pull/3240.

**Comment by beryllw** [comment]
@leonardBang It seems that the failure of unit ci has nothing to do with Paimon's changes.

**Review by leonardBang — APPROVED** [review]
Thanks @beryllw for the update, +1

**Pull request merged**